cli-authpubkey.c

```diff
@@ -266,6 +266,7 @@ int cli_auth_pubkey() {
 		/* Send a trial request */
 		send_msg_userauth_pubkey(key, sigtype, 0);
 		cli_ses.lastprivkey = key;
+		cli_ses.is_trivial_auth = 0;
```

Comment:
It shouldn't clear is_trivial_auth here since it doesn't require any user interaction? For pubkey auth I guess it should clear is_trivial_auth for an agent pubkey signature (not trial request), otherwise leave it set.